tsdb/compact.go

```diff
@@ -232,8 +232,8 @@ func (c *LeveledCompactor) Plan(dir string) ([]string, error) {
 }
 
 func (c *LeveledCompactor) plan(dms []dirMeta) ([]string, error) {
-	slices.SortFunc(dms, func(a, b dirMeta) bool {
-		return a.meta.MinTime < b.meta.MinTime
+	slices.SortFunc(dms, func(a, b dirMeta) int {
+		return int(a.meta.MinTime - b.meta.MinTime)
```

Comment:
Hmm, on a 32bit architecture this will be theoretically wrong as it can lead to overflow. I'm not sure we need to worry about it (@bboreham ?) , but the following is correct regardless of bits
```suggestion
		switch {
		case a.meta.MinTime < b.meta.MinTime:
			return -1
		case a.meta.MinTime > b.meta.MinTime:
			return 1
		default:
			return 0
		}
```

Reply:
@krajorama will you open a PR to change this upstream also @krajorama?

Reply:
The fix makes sense to me FWIW.

Reply:
I'd like to get some feedback from @bboreham , because I might be too paranoid here :(

Reply:
OK. I already committed the suggestion since I thought it makes perfect sense, and prevents you from having to worry about potential overflow.

Reply:
My fix is probably slightly slower, although it shouldn't matter in compactor Plan phase, so it's fine.

Reply:
Change was from @gouthamve I think.
My calculations say we get overflow after 600 hours, which does seem a little close for comfort.

Reply:
Thanks for the input @bboreham - I guess @krajorama's fix is the way to go then?